src/index.js

```diff
@@ -37,22 +37,20 @@ function installFiles(sourceDir, done) {
   switch (npmv.majorVersion()) {
     case '1':
       console.log("[install-files]: WARNING: NPMv1 is not officially supported; unexpected results could occur. Consider upgrading to v2 or later");
+      /* falls through */
     case '2':
       source = sourceDir;
       target = fileInstallingPackagePath && hostPackageDir(fileInstallingPackagePath);
       break;
     case null:
-      console.log("[install-files]: WARNING: Could not determine NPM version"); //Fall back to default
+      console.log("[install-files]: WARNING: Could not determine NPM version");
+      /* falls through */
     default:
       source = path.join(fileInstallingPackagePath, 'node_modules', process.env.npm_package_name, sourceDir);
-      target = fileInstallingPackagePath
+      target = fileInstallingPackagePath;
   }
 
-  if (fileInstallingPackagePath.match(".+" + process.env.npm_package_name + "$")) {
```

Comment:
> a better cross-version solution here would be to check to see if the source and the target package names are the same, and reject if that is the case

Yeah that sounds like it could work!